vaapi/vaapicontext.cpp

```diff
@@ -38,16 +91,13 @@ ConfigPtr VaapiConfig::create(const DisplayPtr& display,
         return ret;
     VAStatus vaStatus;
     VAConfigID config;
-    vaStatus = vaCreateConfig(display->getID(), profile, entry, attribList, numAttribs, &config);
 
-    // VAProfileH264Baseline is super profile for VAProfileH264ConstrainedBaseline
-    // old i965 driver incorrectly claims supporting VAProfileH264Baseline, but not VAProfileH264ConstrainedBaseline
-    if (vaStatus == VA_STATUS_ERROR_UNSUPPORTED_PROFILE
-        && profile == VAProfileH264ConstrainedBaseline
-        && entry == VAEntrypointVLD)
-        vaStatus = vaCreateConfig(display->getID(),
-                                  VAProfileH264Baseline,
-                                  VAEntrypointVLD, attribList, numAttribs, &config);
+    if (!checkProfileCompatible(display, profile)){
```

Comment:
If upper layer want to constrained baseline profile, we won't create main profile unless driver declare main profile supported but constrained baseline unsupported. On the other hand, constrained baseline profile still will be built into bitstream by yami instead of driver. It won't affect decoder.
If we add vld limitation, encoder will fail in such case.